packages/dds/tree/src/domains/json/jsonCursor.ts

```diff
@@ -149,55 +49,42 @@ export class JsonCursor<T> implements ITreeCursor<SynchronousNavigationResult> {
                     return jsonObject.name;
                 }
         }
-    }
-
-    public get keys(): Iterable<FieldKey> {
-        const node = this.currentNode;
-        const type = typeof node;
-
-        switch (type) {
+    },
+    keysFromNode: (node: JsonCompatible): readonly FieldKey[] => {
+        switch (typeof node) {
             case "object":
                 if (node === null) {
                     return [];
                 } else if (Array.isArray(node)) {
                     return [EmptyKey];
                 } else {
-                    return Object.keys(node as object) as Iterable<FieldKey>;
+                    return Object.getOwnPropertyNames(node as object) as FieldKey[];
```

Comment:
Is this inner type conversion needed? Typescript MIGHT be able to infer this from the switch statement;
```suggestion
                    return Object.getOwnPropertyNames(node) as FieldKey[];
```

Also, Object.keys is fine and probably better here. It returns a subset of what getOwnPropertyNames does, since it skips non enumerable properties. (object.keys does NOT return non own properties, the names are misleading: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys )

We don't expect our object to have non enumerable properties (like the length of an array), but if it did, they should probably be skipped (or error). 

```suggestion
                    return Object.keys(node) as FieldKey[];
```